src/main/java/com/fasterxml/jackson/databind/node/FloatNode.java

```diff
@@ -52,6 +52,13 @@
     @Override
     public boolean isFloat() { return true; }
 
+    @Override
+    public boolean hasFractionalPart() {
+        long iPart = (long) _value;
```

Comment:
Ok. Ideally we could do the same as with `double`, but surprisingly, `Math` has fewer method for `float` than `double`. So, two choices I think:

1. Cast to `double`, use `double` check.
2. Check for `isNaN()` of `FloatNode` first (NaNs need to considered to never be really integral, which probably means returning `true` for `hasFractionalPart()` (... but I now realize that name of method might not be optimal; I'll add separate note) and then use `if (_value == Math.round(_value)) { ... is integer, i.e. no fraction...)